examples/crc.rs

```diff
@@ -0,0 +1,76 @@
+//! Example of using the CRC peripheral with various configurations
+//!
+//! A good resource of CRC algorithms, to assist in determining their exact settings,
+//! is https://reveng.sourceforge.io/crc-catalogue/
+
+#![no_main]
+#![no_std]
+
+use cortex_m_rt::entry;
+use log::info;
+
+use stm32h7xx_hal::crc::{Config, Crc, Polynomial};
+use stm32h7xx_hal::{pac, prelude::*};
+
+#[macro_use]
+mod utilities;
+
+/// Standard input data for testing CRCs
+const CHECK_DATA: &[u8] = b"123456789";
+
+#[entry]
+fn main() -> ! {
+    utilities::logger::init();
+    let dp = pac::Peripherals::take().unwrap();
+
+    // Constrain and Freeze power
+    info!("Setup PWR...                  ");
+    let pwr = dp.PWR.constrain();
+    let pwrcfg = example_power!(pwr).freeze();
+
+    // Constrain and Freeze clock
+    info!("Setup RCC...                  ");
+    let rcc = dp.RCC.constrain();
+
+    let ccdr = rcc.sys_ck(200.mhz()).freeze(pwrcfg, &dp.SYSCFG);
+
+    info!("");
+    info!("stm32h7xx-hal example - CRC");
+    info!("");
+
+    let mut crc = Crc::enable(dp.CRC, ccdr.peripheral.CRC);
```

Comment:
Usually we'd expect this to be done via extension traits on the PAC peripheral since this is already how every other peripheral we implemented does it (you can just check any example and the *Ext traits if you do not know what I'm referring to) so just a little inconsistency to the rest of our API.

Reply:
Added extension trait, method is now `dp.CRC.crc(ccdr.peripheral.CRC)`. Wasn't sure if I should name it `crc`, `enable`, or `constrain` but I think this otherwise matches other peripherals.